### ISSUE TITLE
[Feature(s) Request] Delete & Open File Location

### ISSUE THREAD

**Issue by Mihari** [ArturKovacs/emulsion#170]
Two features I would love to see in Emulsion that I sorely miss from even the Windows 10 Photos viewer is the ability to delete an image within the app or go to the images folder location. 

I noticed in [another issue](https://github.com/ArturKovacs/emulsion/issues/131#issuecomment-751856846) that a custom keybind could added to cover a shortcut for printing. Could custom keys be done for these two features as well?

**Comment by ArturKovacs** [comment]
> the ability to delete an image within the app 

Then you are lucky because this is already implemented. If it doesn't work, that's bug 🙁

![Emulsion-del-action](https://user-images.githubusercontent.com/8320264/104807457-87a51a80-57df-11eb-9aa8-ad54239fd839.jpg)

-----------

> go to the images folder location

I assume you mean to open the folder in the file explorer and select the image which emulsion is currently showing. This is something that emulsion can't do currently. There is a fundamental issue with this, namely that it doesn't seem possible to reveal the file in an existing window. This means that every time when revealing the file, a new window has to be opened which clutters your workspace. So let me ask: what is the reason you want to see the file revealed in the folder? The answer to this question could lead us to a better solution to the *real* problem.

With all that said here's the [custom command](https://arturkovacs.github.io/emulsion-website/custom-commands.html) you can use to open the directory and reveal the current file in it.

<details>

<summary> Click to show the cfg entry </summary>

```toml
[[commands]]
input = ["ctrl+r"]
program = "powershell"
args = ["-Command", 'Start-Process -FilePath explorer -ArgumentList "/select, ""${img}"""']
```

</details>

**Comment by Mihari** [comment]
> Then you are lucky because this is already implemented. If it doesn't work, that's bug 🙁
> 

It doesn't work for me, so I suppose it is a bug? Or maybe it just needs to be run as admin, no idea. I should have looked at that keybind list prior more closely, otherwise I would have reported pressing delete didn't seem to work.  

> I assume you mean to open the folder in the file explorer and select the image which emulsion is currently showing. This is something that emulsion can't do currently. There is a fundamental issue with this, namely that it doesn't seem possible to reveal the file in an existing window. This means that every time when revealing the file, a new window has to be opened which clutters your workspace. So let me ask: what is the reason you want to see the file revealed in the folder? The answer to this question could lead us to a better solution to the _real_ problem.

I don't need it to show in an existing window, and your custom command worked exactly as needed and flawlessly, I might also add that powershell immediately closed after the command was issued. So it's not a huge issue on that front for me.

As for the use cases you asked about, as a web dev, I'm constantly looking through folders with tons of files scattered about in the image viewer. Then when I find the one I need, I would tell the viewer to open the file location so I could select it and upload it to the site, or share it with someone else on a messaging platform, etc, etc.

**Comment by ArturKovacs** [comment]
> share it with someone else on a messaging platform

With most messenger applications and websites, this can be done by pressing <kbd>Ctrl</kbd>+<kbd>C</kbd> in Emulsion then <kbd>Ctrl</kbd>+<kbd>V</kbd> in your messaging app. (I should probably describe or demonstrate this on the website)

**Comment by Mihari** [comment]
> > share it with someone else on a messaging platform
> 
> With most messenger applications and websites, this can be done by pressing Ctrl+C in Emulsion then Ctrl+V in your messaging app. (I should probably describe or demonstrate this on the website)

True... As for the other scenario the current work around via command fulfills my need of jumping to the file. So thank you!

One more question while I have you here for features, it's not possible for Emulsion to follow explorer's folder sorting of the files for the purpose of viewing, IE if I have a folder sorted by file date, in Windows Photos it respected the sorting and could go through them in that order. I'm going to assume that's not possible, right? Especially given the multi-platform nature of Emulsion.

**Comment by ArturKovacs** [comment]
> fulfills my need of jumping to the file. So thank you!

You're welcome. Also I realized that the pesky powershell window can be entirely avoided by using the following configuration

```toml
[[commands]]
input = ["ctrl+r"]
program = "explorer"
args = ["/select,", "${img}"]
```

------

> I have a folder sorted by file date, in Windows Photos it respected the sorting and could go through them in that order

Hmm this is indeed true. I'm almost certain that if Photos can do this then all other applications can too, so in theory it should be possible. After quite some time searching online I could not find anything that seemed helpful so the information on how this can be done might be burried somewhere in the Microsoft online docs. I opened #171
